### PR TITLE
Pr/api set platform

### PR DESCRIPTION
**Pull request by hunger**
Rename `set_platform_abstraction` to `set_platform` and `PlatformAbstraction` to `Platform`.

Fix some typos reported by cspell in follow up patches.

Fixes: #1503, #1536

### CODE REVIEW

File: internal/core/animations.rs [hunk 1]
```diff
@@ -206,7 +206,7 @@ impl Instant {
     }
 
     fn duration_since_start() -> core::time::Duration {
-        crate::platform::PLAFTORM_ABSTRACTION_INSTANCE
+        crate::platform::PLATFORM_ABSTRACTION_INSTANCE
```
Comment (tronical) [inline internal/core/animations.rs:209]: 😂

**Review by tronical — APPROVED** [review]
I suggest to also cc the corresponding GitHub Issue that this partially fixes.

However I do think for this to pass the CI, you're also going to have to change the call sites in the `mcu-board-support` crate.

File: api/rs/slint/Cargo.toml [hunk 2]
```diff
@@ -19,7 +19,13 @@ path = "lib.rs"
 
 [features]
 
-default = ["std", "eventloop-winit", "renderer-femtovg", "backend-qt", "compat-0-2-0"]
+default = [
```
Comment (tronical) [inline api/rs/slint/Cargo.toml:22]: Did you change anything in this file or are you just reformatting it?

Comment (hunger) [reply api/rs/slint/Cargo.toml:22]: Yes, this file got added due to renaming `PlatformAbstraction` to `Platform` in line 78.

**Pull request merged**